### ISSUE TITLE
Include failure conditions in docstrings

### ISSUE THREAD

**Issue by 11xor6** [gtsystem/lightkube#72]
You've done a great job with this library and one of the things I most appreciate is the extensive documentation I can get within my editor, including the type hints. However the one thing I find to be missing is the failure conditions. For example what happens when I call `Client.get(...)` for an object that doesn't exist? Does it return an empty object? None? Throw an exception? The docstring doesn't specify. This means you need to dig into the code to find out what it does and which exceptions it throws. It'd honestly be great if the docstrings specified such conditions.

**Comment by gtsystem** [comment]
I agree with you, this will be nice to have. If you would like to contribute this change it will great.
